examples/question_answering/training_example.py

```diff
@@ -0,0 +1,45 @@
+from datasets import load_dataset
+import csv
+from happytransformer.happy_question_answering import  HappyQuestionAnswering
+
+
+def main():
+    # Be careful not to commit the csv files to the rep
+    train_csv_path = "train.csv"
+    eval_csv_path = "eval.csv"
+
+    train_dataset = load_dataset('squad', split='train')
+    eval_dataset = load_dataset('squad', split='validation')
+
+    generate_csv(train_csv_path, train_dataset, 500)
+    generate_csv(eval_csv_path, eval_dataset, 100)
+
+    happy_qa = HappyQuestionAnswering(model_type="BERT", model_name="bert-base-uncased")
+    before_loss = happy_qa.eval(eval_csv_path)
+    happy_qa.train(train_csv_path)
+    after_loss = happy_qa.eval(eval_csv_path)
+
+    print("Before loss: ", before_loss)
+    print("After loss: ", after_loss)
+    assert after_loss < before_loss
+
+
+def generate_csv(csv_path, dataset, count):
+    with open(csv_path, 'w', newline='') as csvfile:
+        writter = csv.writer(csvfile)
+        writter.writerow(["context", "question", "answer_text", "answer_start"])
+        i = 0
+        for case in dataset:
+            context = case["context"]
+            question = case["question"]
+            answer_text = case["answers"]["text"][0]
+            answer_start = case["answers"]["answer_start"][0]
+            writter.writerow([context, question, answer_text, answer_start])
```

Comment:
Could we use a `DictWriter` here?

Reply:
I don't think so because "case" is a dictionary that contains dictionaries within it. Then, some of the sub dictionaries have lists. 

Unless you mean converting  context, question, answer_text, answer_start into a dictionary. But I don't think that's any better than what we have now.

Reply:
> Unless you mean converting context, question, answer_text, answer_start into a dictionary. But I don't think that's any better than what we have now.

that is what I'm describing. It would reduce assignment and ensure consistency